src/main/kotlin/com/safehill/kclient/controllers/ConversationThreadController.kt

```diff
@@ -0,0 +1,40 @@
+package com.safehill.kclient.controllers
+
+import com.safehill.kclient.models.interactions.InteractionAnchor
+import com.safehill.kclient.models.users.ServerUser
+import com.safehill.kclient.network.ServerProxy
+import com.safehill.kclient.util.runCatchingPreservingCancellationException
+
+class ConversationThreadController(
+    val serverProxy: ServerProxy,
+    val userInteractionController: UserInteractionController,
+    val encryptionDetailsController: EncryptionDetailsController
+) {
+
+    suspend fun provideEncryptionDetailToUsers(
+        users: List<ServerUser>,
+        threadId: String
+    ): Result<Unit> {
+        return runCatchingPreservingCancellationException {
+            val symmetricKey = userInteractionController.getSymmetricKey(
+                anchorId = threadId,
+                interactionAnchor = InteractionAnchor.THREAD
+            )
+            if (symmetricKey == null) {
+                throw UserInteractionController.InteractionErrors.MissingE2EEDetails(
+                    anchorId = threadId,
+                    anchor = InteractionAnchor.THREAD
+                )
+            } else {
+                serverProxy.provideEncryptionDetails(
```

Comment:
Same comment as in the other PR. The verb "provide" has a specific meaning and we shouldn't use it in this context. Can you please align the SafehillAPI method names between Safehill-Client (Swift) and Safehill-kClient (Kotlin)?

Reply:
can you help me point to the file in Safehill-Client (Swift) that contains the functions?